docs/dev-guide/models.d2

```diff
@@ -0,0 +1,138 @@
+# Node styles
+users.style.fill: "#ffe4e1"
+administrators.style.fill: "#ffe4e1"
+pl_courses: "pl_courses*" {
+  style.fill: "#ffe4e1"
+}
+course_permissions.style.fill: "#ffe4e1"
+
+courses.style.fill: lemonchiffon
+exams.style.fill: lemonchiffon
+reservations.style.fill: lemonchiffon
+
+course_instances: "course_instances*" {
+  style.fill: "#f0fff0"
+}
+course_instance_access_rules.style.fill: "#f0fff0"
+assessment_access_rules.style.fill: "#f0fff0"
+questions: "questions*" {
+  style.fill: "#f0fff0"
+}
+question_tags.style.fill: "#f0fff0"
+tags.style.fill: "#f0fff0"
+assessments: "assessments*" {
+  style.fill: "#f0fff0"
+}
+assessment_questions: "assessment_questions*" {
+  style.fill: "#f0fff0"
+}
+assessment_sets.style.fill: "#f0fff0"
+topics.style.fill: "#f0fff0"
+alternative_groups.style.fill: "#f0fff0"
+zones.style.fill: "#f0fff0"
+
+issues.style.fill: lavender
+enrollments.style.fill: lavender
+assessment_instances.style.fill: lavender
+instance_questions.style.fill: lavender
+variants.style.fill: lavender
+submissions.style.fill: lavender
+grading_jobs.style.fill: lavender
+job_sequences.style.fill: lavender
+jobs.style.fill: lavender
+
+assessment_state_logs.style.fill: white
+assessment_score_logs.style.fill: white
+question_score_logs.style.fill: white
+variant_view_logs.style.fill: white
+assessment_view_logs.style.fill: white
+
+# Connections
+administrators -> users
+course_permissions -> users
+course_permissions -> pl_courses
+
+courses -> pl_courses
+exams -> courses
+reservations -> users
+reservations -> exams
+
+assessment_access_rules -> assessments
+course_instances -> pl_courses
+course_instance_access_rules -> course_instances
+questions -> pl_courses
+questions -> topics
+question_tags -> questions
+question_tags -> tags
+tags -> pl_courses
+assessments -> course_instances
+assessments -> assessment_sets
+assessment_questions -> assessments
+assessment_questions -> alternative_groups
+assessment_questions -> questions
+assessment_sets -> pl_courses
+topics -> pl_courses
+zones -> assessments
+alternative_groups -> zones
+
+issues -> variants: {
+  style.stroke-dash: 5
+}
+enrollments -> users
+enrollments -> course_instances
+assessment_instances -> assessments
+assessment_instances -> users
+instance_questions -> assessment_questions
+instance_questions -> assessment_instances
+variants -> questions
+variants -> users
+variants -> instance_questions: {
+  style.stroke-dash: 5
+}
+submissions -> variants
+job_sequences -> pl_courses
+job_sequences -> users
+jobs -> job_sequences
+
+assessment_state_logs -> assessment_instances
+grading_jobs -> submissions
+assessment_score_logs -> assessment_instances
+question_score_logs -> instance_questions
+question_score_logs -> grading_jobs: {
+  style.stroke-dash: 5
+}
+variant_view_logs -> variants
+assessment_view_logs -> assessment_instances
+
+legend: "Legend"{
+    near: center-left
+    grid-columns: 1
+
+  k5: PrairieSchedule tables. {
+    style.fill: lemonchiffon
+  }
```

Comment:
Nit: there's some inconsistent spacing here. It looks like d2 ships with an autoformatter. We _really_ like autoformatters; can we run that in CI to enforce standardized formatting?

Reply:
I added some CI to the makefile.

Reply:
Let me know if this is sufficient, d2 fmt doesn't have any stdout so it was a little annoying.

Reply:
Formatted all.